src/waltham/waltham-connection.h

```diff
@@ -373,6 +354,83 @@ wth_connection_get_protocol_error(struct wth_connection *conn,
 				  const char **interface,
 				  uint32_t *object_id);
 
+/** Return a registry using the internal wth_display
+ *
+ * \param conn The Waltham connection.
+ * \return The registry obtained.
+ *
+ * The returned registry can then be used to retrieve other global
+ * objects, using its usual interface.
+ *
+ * \memberof wth_connection
+ * \client_api
+ */
+struct wthp_registry *
+wth_connection_get_registry(struct wth_connection *conn);
+
+/** Return a callback using the internal wth_display
+ *
+ * \param conn The Waltham connection.
+ * \return The callback obtained.
```

Comment:
Likewise, maybe document this as "\return A new callback object".
Below, "the related request" is a little vague. How about "once the server has seen all preceding requests"?

Reply:
I mostly copy/pasted this comment from the protocol file IIRC, maybe we should fix that in both Wayland and Waltham XML files then.  I will also fix this one comment.

Reply:
Sure.